pkg/controller/deployment/deployment_controller.go

```diff
@@ -1176,11 +1294,15 @@ func (dc *DeploymentController) scaleDownOldReplicaSetsForRecreate(oldRSs []*ext
 
 // scaleUpNewReplicaSetForRecreate scales up new replica set when deployment strategy is "Recreate"
 func (dc *DeploymentController) scaleUpNewReplicaSetForRecreate(newRS *extensions.ReplicaSet, deployment *extensions.Deployment) (bool, error) {
+	glog.V(4).Infof("Scaling up new replica set %q from %d to %d", newRS.Name, newRS.Spec.Replicas, deployment.Spec.Replicas)
 	scaled, _, err := dc.scaleReplicaSetAndRecordEvent(newRS, deployment.Spec.Replicas, deployment)
 	return scaled, err
 }
 
 func (dc *DeploymentController) cleanupOldReplicaSets(oldRSs []*extensions.ReplicaSet, deployment *extensions.Deployment) error {
+	if deployment.Spec.RevisionHistoryLimit == nil {
```

Comment:
> Not really. This has always been the behavior for Deployments prior to adding RevisionHistoryLimit. 

Seems scary to me.  From a human perspective a history limit of 1000 isn't that different from infinite, but a buggy component could to 50K deployment revisions per week and maybe no one would notice.  Seems a lot safer to just put a default of "more than anyone should want, but going to start GCing before it starts causing issues".